std/conv.d

```diff
@@ -207,6 +207,13 @@ template to(T)
     {
         return toImpl!T(arg);
     }
+
+    // Fix issue 16108
+    T to(S)(ref S arg)
+        if (isAggregateType!S && !isCopyable!S)
+    {
+        return toImpl!T(arg);
+    }
 }
```

Comment:
Ah, sorry I didn't see your proposed template signature and misunderstood.

Well, you definitely never want to pass a static array by value, as that would cause a lot of the code to return references to stack data. That's why the original special case was added.